### DISCUSSION TITLE
How to ensure that selected root-path prefix is included in the service-desc document

### DISCUSSION THREAD

**Discussion by jfbourgon** [stac-utils/stac-fastapi#489] [Q&A] [answered]
We deployed STAC-fastapi behind a proxy and followed the uvicorn/fastapi documentation in other to have the URL paths rendered by the application including the selected root path. 

Basically our application is started with something like the following:
```
uvicorn --root-path "/stac" --proxy-headers stac_fastapi.sqlalchemy.app:app
```

All the links rendered by the application are the ones expected and they can be used to reach the various endpoints supported by the application. The only notable exception is inside the service-desc located under `/stac/api`, this endpoint is reachable but its content provides broken links since none of them include the expected root-path prefix (`/stac` in our specific case).

**Current**
```
{
	"openapi": "3.0.2",
	"info": {
		"title": "stac-fastapi",
		"version": "0.1"
	},
	"paths": {
		"/": {
			"get": {
				"summary": "Landing Page",
				"description": "Endpoint.",
				"operationId": "Landing_Page__get",
				"responses": {
					"200": {
						"description": "Successful Response",
						"content": {
							"application/json": {
								"schema": {}
							}
						}
					}
				}
			}
		},
		"/conformance": {
			"get": {
				"summary": "Conformance Classes",
				"description": "Endpoint.",
				"operationId": "Conformance_Classes_conformance_get",
				"responses": {
					"200": {
						"description": "Successful Response",
						"content": {
							"application/json": {
								"schema": {}
							}
						}
					}
				}
			}
		},
		"/collections/{collection_id}/items/{item_id}": {
			"get": {
				"summary": "Get Item",
				"description": "Endpoint.",
				"operationId": "Get_Item_collections__collection_id__items__item_id__get",
				"parameters": [
...
```

**Expected**
```
{
	"openapi": "3.0.2",
	"info": {
		"title": "stac-fastapi",
		"version": "0.1"
	},
	"paths": {
		"/stac/": {
			"get": {
				"summary": "Landing Page",
				"description": "Endpoint.",
				"operationId": "Landing_Page__get",
				"responses": {
					"200": {
						"description": "Successful Response",
						"content": {
							"application/json": {
								"schema": {}
							}
						}
					}
				}
			}
		},
		"/stac/conformance": {
			"get": {
				"summary": "Conformance Classes",
				"description": "Endpoint.",
				"operationId": "Conformance_Classes_conformance_get",
				"responses": {
					"200": {
						"description": "Successful Response",
						"content": {
							"application/json": {
								"schema": {}
							}
						}
					}
				}
			}
		},
		"/stac/collections/{collection_id}/items/{item_id}": {
			"get": {
				"summary": "Get Item",
				"description": "Endpoint.",
				"operationId": "Get_Item_collections__collection_id__items__item_id__get",
				"parameters": [
...
```

This omission makes all the potential interactions with the API available inside the Swagger rendering with `Try it out` button unusable.

Any advice on how to achieve this by configuration and/or links to the relevant documentation from the various components (STAC-fastapi/FastAPI/Uvicorn) are more than welcome.

**Comment by geospatial-jeff** [comment, marked as answer]
Hi @jfbourgon  please check out the planetary computer source code for an example of how to do this, their docs endpoint appears to route to the correct endpoints and `Try it out` buttons are working (https://github.com/microsoft/planetary-computer-apis/blob/main/pcstac/pcstac/api.py#L22-L26).

Running behind a proxy is common enough it might make sense to pull this into stac-fastapi.

**Reply by jfbourgon** [reply]
Thank for the suggestion. The custom StacAPI class would work. If you agree that this requirement is common enough to have it also implemented into stac-fastapi, I can create the issue for this feature.

**Reply by jfbourgon** [reply]
created #497 in order to have this feature implemented inside stac-fastapi
